llvm/lib/Transforms/Scalar/Reassociate.cpp

```diff
@@ -437,7 +437,8 @@ static bool LinearizeExprTree(Instruction *I,
     for (unsigned OpIdx = 0; OpIdx < I->getNumOperands(); ++OpIdx) { // Visit operands.
       Value *Op = I->getOperand(OpIdx);
       LLVM_DEBUG(dbgs() << "OPERAND: " << *Op << " (" << Weight << ")\n");
-      assert(!Op->use_empty() && "No uses, so how did we get to it?!");
+      assert((isa<ConstantData>(Op) || !Op->use_empty()) &&
```

Comment:
Why do we need this change? use_empty() should still work on ConstantData?

Reply:
I've started thinking maybe it shouldn't. The hasN* use cases don't really mean anything useful given the references are context wide